src/parser/type_checking.c

```diff
@@ -7809,12 +7809,15 @@ pt_fold_constants_pre (PARSER_CONTEXT * parser, PT_NODE * node, void *arg, int *
   switch (node->node_type)
     {
     case PT_FUNCTION:
-      if (node->info.function.function_type == F_BENCHMARK)
+    case PT_EXPR:
+      if (node->node_type == PT_FUNCTION)
 	{
-	  // we want to test full execution of sub-tree; don't fold it!
-	  *continue_walk = PT_LIST_WALK;
+	  if (node->info.function.function_type == F_BENCHMARK)
+	    {
+	      // we want to test full execution of sub-tree; don't fold it!
+	      *continue_walk = PT_LIST_WALK;
+	    }
```

Comment:
if (node->info.function.function_type == F_BENCHMARK)
    {
	  // we want to test full execution of sub-tree; don't fold it!
	  *continue_walk = PT_LIST_WALK;
     }
    [[fallthrough]];

    case PT_EXPR: